packages/flutter_tools/lib/src/context_runner.dart

```diff
@@ -214,6 +217,10 @@ Future<T> runInContext<T>(
         platform: globals.platform,
       ),
       ProcessInfo: () => ProcessInfo(),
+      ProcessManager: () => ErrorHandlingProcessManager(
```

Comment:
If you mirror the way the `ErrorHandlingFileSystem` is set up:

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/globals.dart#L66

then tests that inject their own `ProcessManager` should also get wrapped in the `ErrorHandlingProcessManager`.

Reply:
That seems like the opposite of what I want - if a test specifies a regular process manager it should get a regular process manager. This would match the behavior of the non-context tests

Reply:
If we have tests with a mock process manager where the mock process manager throws a `ProcessException` with one of the errors now caught by the `ErrorHandlingProcessManager`, those tests are now testing something that can't actually happen.

Reply:
that's true for context injected tests, but I don't think that we should require all of the context free tests to use the error handling filesystem/process manager.

testing cases that can't happen is definitely not idea - we should be able to handle this in code review by pointing out when something is redundant with the error_handling_io lib.

I'm more worried about tests which build up a large amount of injected interfaces being hard to understand for new contributors. Why does this process exception become a tool exit here, but not here? Or what instance am I actually using.